src/CallSettings.php

```diff
@@ -217,30 +191,34 @@ public function getUserHeaders()
     public function merge(CallSettings $otherSettings = null)
     {
         if (is_null($otherSettings)) {
-            return new CallSettings([
-                'timeoutMillis' => $this->timeoutMillis,
-                'retrySettings' => $this->retrySettings,
-                'userHeaders' => $this->userHeaders,
-            ]);
-        } else {
-            $timeoutMillis = $this->timeoutMillis;
-            if ($otherSettings->getTimeoutMillis() != self::INHERIT_TIMEOUT) {
-                $timeoutMillis = $otherSettings->getTimeoutMillis();
-            }
-            $retrySettings = $this->retrySettings;
-            if (is_null($otherSettings->getRetrySettings())
-                || !$otherSettings->getRetrySettings()->shouldInherit()) {
-                $retrySettings = $otherSettings->getRetrySettings();
-            }
-            $userHeaders = $this->userHeaders;
-            if (!is_null($otherSettings->userHeaders)) {
-                $userHeaders = $otherSettings->userHeaders;
-            }
-            return new CallSettings([
-                'timeoutMillis' => $timeoutMillis,
-                'retrySettings' => $retrySettings,
-                'userHeaders' => $userHeaders,
-            ]);
+            return $this->with([]);
+        }
+        return $this->with($otherSettings->toArray());
+    }
+
+    /**
+     * @return array
+     */
+    private function toArray()
+    {
+        $arr = [];
+        if (!is_null($this->getRetrySettings())) {
+            $arr['retrySettings'] = $this->getRetrySettings();
+        }
+        if (!is_null($this->getUserHeaders())) {
+            $arr['userHeaders'] = $this->getUserHeaders();
+        }
+        return $arr;
+    }
+
+    private static function convertArrayFromSnakeCase($settings)
+    {
+        $camelCaseSettings = [];
+        foreach ($settings as $key => $value) {
+            $wordsKey = str_replace('_', ' ', $key);
+            $camelCaseKey = str_replace(' ', '', ucwords($wordsKey));
```

Comment:
It seems like you can just specify `'_'` as a delimiter in [ucwords](http://php.net/manual/en/function.ucwords.php), so the line 218 will become redundant.

Reply:
This will work:

```php
str_replace('_', '', ucwords($key, '_'))
```